### PR TITLE
Added support for SSHA password hashing

### PR DESCRIPTION
**Pull request by jwsdv**

### CODE REVIEW

File: pam_mysql.c [hunk 1]
```diff
@@ -632,9 +636,66 @@ static char *pam_mysql_md5_data(const unsigned char *d, unsigned int sz, char *m
 #endif
 /* }}} */
 
-/* {{{ pam_mysql_sha1_data */
 #if defined(HAVE_OPENSSL)
 #define HAVE_PAM_MYSQL_SHA1_DATA
+/* {{{ calcDecodeLength */
```
Comment (NigelCunningham) [inline pam_mysql.c]: Could I get you to put in 'proper' documentation? (Ala PHPdocs / kernel docs) I've started to do this for other functions in the file as you'll see from last night (Australian time)'s commits.

File: pam_mysql.c [hunk 2]
```diff
@@ -3023,6 +3122,30 @@ static pam_mysql_err_t pam_mysql_check_passwd(pam_mysql_ctx_t *ctx,
...
+              vresult = strcmp(row[0], buf);
+              {
+                char *p = buf - 1;
+                while (*(++p)) *p = '\0';
+              }
+#else
+              syslog(LOG_AUTHPRIV | LOG_ERR, PAM_MYSQL_LOG_PREFIX "non-crypt()ish SSHA hash is not supported in this build.");
+#endif
+            } break;
+
```
Comment (NigelCunningham) [inline pam_mysql.c]: Looks good. Could I also get you to provide (in a comment for now) a couple of test scenarios? I plan on adding unit tests soon. Thanks for the work!

File: pam_mysql.c (hunk 1 above)
Comment (juergenwsdv) [inline pam_mysql.c]: Done.

File: pam_mysql.c (hunk 2 above)
Comment (juergenwsdv) [inline pam_mysql.c]: This is analogous to SHA1. Please provide an example of the test scenario for SHA1, so I can customize it for SSHA.

Comment (NigelCunningham) [inline pam_mysql.c]: Oh - don't get too carried away; I just want to have a couple of sample 'real world' arguments and the results that should be returned. I'll make it into a 'proper' test once I get to that point.

**Comment by jwsdv** [comment]
I have closed this pull request and added a new one to merge into v0.8.x

**Pull request closed without merging**